cmd/flux/cmd/test.go

```diff
@@ -0,0 +1,151 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/edit"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/fluxinit"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/runtime"
+	"github.com/spf13/cobra"
+)
+
+type Test struct {
+	ast *ast.Package
+	Err error
+}
+
+func NewTest(ast *ast.Package) Test {
+	return Test{
+		ast: ast,
+	}
+}
+
+func (t *Test) Name() string {
+	return t.ast.Files[0].Name
+}
+
+func (t *Test) Run() {
+	jsonAST, Err := json.Marshal(t.ast)
+	if Err != nil {
+		t.Err = Err
+		return
+	}
+	c := lang.ASTCompiler{AST: jsonAST}
+
+	ctx := executetest.NewTestExecuteDependencies().Inject(context.Background())
+	program, Err := c.Compile(ctx, runtime.Default)
+	if Err != nil {
+		t.Err = errors.Wrap(Err, codes.Invalid, "failed to compile")
+	}
+
+	alloc := &memory.Allocator{}
+	result, Err := program.Start(ctx, alloc)
+	if Err != nil {
+		// XXX: rockstar (8 Dec 2020) - Not all tests should return streaming data.
+		if !strings.Contains(Err.Error(), "this Flux script returns no streaming data") {
```

Comment:
I wonder if there's a way we can use the new errors API to deal with this. I'd prefer not to use `strings.Contains` on the error message if possible. Can you replace this with a method to hide the usage of `strings.Contains` so it's easy for us to experiment with this?

I'd like to experiment with something like using `errors.Unwrap` to find the appropriate error or `errors.Is`.

Reply:
I also don't like this spelling, and am encouraged by the `errors.Unwrap` and `errors.Is`, but I couldn't really find a way to discover what I needed with those. I'm also not a huge fan of encapsulating functionality that's only used once inside another function. At best, it requires a push onto my mental stack. Experimenting in an encapsulated function or right here would result in the same amount of lines changed. Thoughts?

Reply:
Let's just leave it for now. It's not too hard to extract this into a function when/if we experiment.

Reply:
Glancing at it now, I forgot that our flux scripts still all have to return streaming data. I tried to make that not a requirement, but the Flux APIs don't really lend themselves to it currently. As such, I just removed the logic entirely, and when we come back to support tests that don't return streaming data, we can implement this functionality more fully.